app/services/hyrax/collections/permissions_service.rb

```diff
@@ -1,77 +1,118 @@
 module Hyrax
   module Collections
     class PermissionsService
+      # @api private
+      #
+      # IDs of collections, including admin sets, a user can access based on participant roles.
+      #
+      # @param user [User] user
+      # @param access [Array<String>] one or more types of access (e.g. Hyrax::PermissionTemplateAccess::MANAGE, Hyrax::PermissionTemplateAccess::DEPOSIT, Hyrax::PermissionTemplateAccess::VIEW)
+      # @return [Array<String>] IDs of collections for which the user has specified roles
```

Comment:
NOTE: It is preferable in the access query methods to return a list of IDs instead of the collection objects.  The IDs can be combined with solr search builders to efficiently retrieve the additional information needed for display in the UI.  Instantiating the collection objects from Fedora is less efficient, but can be done from the IDs if needed.